### ISSUE TITLE
JSON encoder emits invalid strings

### ISSUE THREAD

**Issue by pjt33** [telefonicaid/fiware-orion#1172]
If I create an attribute with characters which need escaping in JSON strings:

	{
		"updateAction": "APPEND",
		"contextElements": [
			{
				"id": "Bug2",
				"isPattern": "false",
				"type": "Bug",
				"attributes": [
					{
						"name": "foo",
						"type": "string",
						"value": "line1\nline2"
					}
				]
			}
		]
	}

then when Orion encodes it in JSON for query responses or to notify subscribers it fails to escape:

	{
	  "contextElement" : {
		"type" : "Bug",
		"isPattern" : "false",
		"id" : "Bug2",
		"attributes" : [
		  {
			"name" : "foo",
			"type" : "string",
			"value" : "line1
	line2"
		  }
		]
	  },
	  "statusCode" : {
		"code" : "200",
		"reasonPhrase" : "OK"
	  }
	}

**Comment by kzangeli** [comment]
Not sure that the broker should modify the values of the fields that is input to the broker.
What if you want to store e.g. a metadata as "newline": "\n" and then the broker goes and changes it to "\\n" ?
We will look into it, of course, but right now I don't really see how the broker could modify the data ...

**Comment by pjt33** [comment]
I don't think it is modifying the data as such. I think it's correctly handling escaping on input but not on output.

**Comment by fgalan** [comment]
Actually, from a JSON point fo view the followint two JSONs:

```{
    "updateAction": "APPEND",
    "contextElements": [
        {
            "id": "Bug2",
            "isPattern": "false",
            "type": "Bug",
            "attributes": [
                {
                    "name": "foo",
                    "type": "string",
                    "value": "line1\nline2"
                }
            ]
        }
    ]
}
```

and

```
{
    "updateAction": "APPEND",
    "contextElements": [
        {
            "id": "Bug2",
            "isPattern": "false",
            "type": "Bug",
            "attributes": [
                {
                    "name": "foo",
                    "type": "string",
                    "value": "line1
line2"
                }
            ]
        }
    ]
}
```

are the same. Thus, under my understanding, Orion is behaving correctly.

**Comment by pjt33** [comment]
The [JSON spec](http://json.org/) specifically prohibits control characters (which includes literal newlines) from strings, requiring them to be escaped as `\n`. Orion is not escaping, and so it is not producing conforming JSON. My (third party) consumer does not have an option to enable processing of non-conforming input.

**Comment by fgalan** [comment]
We will have a closer look to the JSON spec in order to ensure that Orion is behaving correctly and fix if it is something wrong. Thanks for the report!

**Comment by fgalan** [comment]
Fixed in PR #1626 (release/0.26.1) and PR #1627 (develop).
